### ISSUE TITLE
Add GitHub Actions in CI

### ISSUE THREAD

**Issue by echarles** [jupyter/jupyter_server#143]
We have for now travis and appveyor for the CI. Travis shows to be a bit flaky, while my experience with GitHub actions is very good.

What about replacing Travis with GitHub Actions?

**Comment by rolweber** [comment]
If somebody volunteers to implement tests with GitHub Actions, it certainly wouldn't hurt to have those running in parallel with Travis for a while. The decision whether to disable one or the other can be deferred until we know how well both are working, in direct comparison.

**Comment by maartenbreddels** [comment]
Yes, I've done the same with [vaex](https://github.com/vaexio/vaex) where GitHub actions give me fast feedback, but I'm keeping Travis/AppVeyor around for a while.

**Comment by kevin-bates** [comment]
@Zsailer has been reworking the tests (moving to pytest and using GH actions): https://github.com/Zsailer/jupyter_server_tests

I agree it would be good to keep Travis and appveyor in the mix until we can confirm GHA meets our needs.

**Comment by echarles** [comment]
Done with #146
